_posts/2015-08-11-postgres-remove-duplicates.md

```diff
@@ -0,0 +1,168 @@
+---
+layout:     post
+title:      Removing duplicates in Postgres
+date:       2015-08-11 23:00:00
+summary:    This is a study to find ways in which we can remove duplicate rows
+            based on a certain criteria from our tables.
+categories: sql
+---
+
+The best way to avoid duplicates in our database is to use a `UNIQUE` constraint
+or index in the first place, but as we know sometimes things get out of control
+and some housekeeping is needed. Let's see some ways to do this:
+
+Suppose we have a table called orders like the following:
+
+{% highlight sql %}
+CREATE TABLE orders(
+  id serial PRIMARY KEY,
+  item_id integer,
+  code varchar
+);
+
+INSERT INTO orders(item_id, code)
+VALUES (1, 'abc'),
+       (1, 'abc'),
+       (2, 'dba'),
+       (2, 'hjk'),
+       (2, 'dba'),
+       (3, 'fkl');
+
+SELECT * FROM orders;
+
+ id | item_id | code
+----+---------+------
+  1 |       1 | abc
+  2 |       1 | abc
+  3 |       2 | dba
+  4 |       2 | hjk
+  5 |       2 | dba
+  6 |       3 | fkl
+(6 rows)
+{% endhighlight %}
+
+We want to remove rows that have the same `item_id` and `code`, what are our
+options? Actually, a lot! (If your table doesn't  have a unique identifier it's
+useful to read [this](#ctid) section first.)
```

Comment:
lol.. I think the `anchor` confused me.. Maybe you can start this information here as a section necessary to the other.

Regarding the `anchor` strategy, I think the markdown has something for this (you don't need to create the `a` tag)